### PR TITLE
Simplify control flow, deduplicate code

### PR DESCRIPTION
**Pull request by andralex**
Following discussion around this particular code.

### CODE REVIEW

File: src/scanelf.d [hunk 1]
```diff
@@ -52,27 +51,45 @@ void scanElfObjModule(void delegate(const(char)[] name, int pickAny) pAddSymbol,
...
         }
+        else
+        {
+            static assert(model == 64);
+            alias ElfXX_Ehdr = Elf64_Ehdr;
+            alias ElfXX_Shdr = Elf64_Shdr;
+            alias ElfXX_Sym = Elf64_Sym;
+        }
+
+        if (base.length < ElfXX_Ehdr.sizeof)
```
Comment (andralex) [inline src/scanelf.d]: btw this check was not done for 32 bit. Was that intentional?

Comment (WalterBright) [inline src/scanelf.d]: It was there in the original code line 51

**Comment by andralex** [comment]
it started passing :)

**Comment by WalterBright** [comment]
Auto-merge toggled on

**Pull request merged**